File: stepik-java-api/src/main/java/org/stepik/api/queries/auth/CodeAuthenticationPostQuery.java
package org.stepik.api.queries.auth;

import org.jetbrains.annotations.NotNull;
import org.stepik.api.actions.StepikAbstractAction;
import org.stepik.api.auth.GrantTypes;


public class CodeAuthenticationPostQuery extends AbstractAuthorizationPostQuery {
    public CodeAuthenticationPostQuery(@NotNull StepikAbstractAction stepikAction) {
        super(stepikAction);
        addParam("grant_type", GrantTypes.AUTHORIZATION_CODE.toString());
    }

    @NotNull
    public CodeAuthenticationPostQuery clientId(@NotNull String clientId) {
        addParam("client_id", clientId);
        return this;
    }

    @NotNull
    public CodeAuthenticationPostQuery code(@NotNull String code) {
        addParam("code", code);
        return this;
    }

    @NotNull
    public CodeAuthenticationPostQuery redirectUri(@NotNull String redirectUri) {
        addParam("redirect_uri", redirectUri);
        return this;
    }
}

File: stepik-java-api/src/main/kotlin/org/stepik/api/queries/auth/CodeAuthenticationPostQuery.kt
package org.stepik.api.queries.auth

import org.stepik.api.actions.StepikAbstractAction
import org.stepik.api.auth.GrantTypes

class CodeAuthenticationPostQuery(stepikAction: StepikAbstractAction) : AbstractAuthorizationPostQuery(stepikAction) {
    init {
        addParam("grant_type", GrantTypes.AUTHORIZATION_CODE.toString())
    }
    
    fun clientId(clientId: String): CodeAuthenticationPostQuery {
        addParam("client_id", clientId)
        return this
    }
    
    fun code(code: String): CodeAuthenticationPostQuery {
        addParam("code", code)
        return this
    }
    
    fun redirectUri(redirectUri: String): CodeAuthenticationPostQuery {
        addParam("redirect_uri", redirectUri)
        return this
    }
}
